scalyr_agent/builtin_monitors/journald_monitor.py

```diff
@@ -435,3 +563,129 @@ def gather_sample(self):
             self._checkpoint.update_checkpoint(
                 self._checkpoint_name, str(self._last_cursor)
             )
+
+    def stop(self, wait_on_join=True, join_timeout=5):
+        ScalyrMonitor.stop(self, wait_on_join=wait_on_join, join_timeout=join_timeout)
+        self.log_manager.close()
+        self._logger.info("Stop was called on the journald monitor")
+
+
+class LogConfigManager:
+    def __init__(
+        self,
+        global_config,
+        formatter,
+        max_log_size=20 * 1024 * 1024,
+        max_log_rotations=2,
+        test_logger=None,
+    ):
+        self._loggers = {}
+        self._max_log_size = max_log_size
+        self._max_log_rotations = max_log_rotations
+        self._global_config = global_config
+        self._formatter = formatter
+        self.log_watcher = None
+        self.test_logger = test_logger
+
+        self.__log_config_creators = self.initialize()
+
+    def initialize(self):
+        """ TODO: document
+        """
+        config_matchers = []
+        catchall_defined = False
+        for config in self._global_config.journald_log_configs:
+            config_matcher = self.create_config_matcher(config)
+            config_matchers.append(config_matcher)
+            if config["journald_unit"] == ".*":
+                catchall_defined = True
+        if not catchall_defined:
+            config_matchers.append(self.create_config_matcher({}))
+
+        return config_matchers
+
+    def create_config_matcher(self, conf):
+        """ TODO: document
+        """
+        config = conf.copy()
```

Comment:
This only does a shallow copy.. Let's do a deep one just in case.  We don't really need it now.. but I don't want us to get bit by this latter on if the code changes and we do actually need one.